src/Module.zig

```diff
@@ -2515,6 +2525,17 @@ pub const SrcLoc = struct {
     }
 };
 
+/// This wraps a simple integer in debug builds so that later on we can find out
+/// where in semantic analysis the value got set.
+const TracedOffset = struct {
+    x: i32,
+    trace: Trace = trace_init,
+
+    const want_tracing = builtin.mode == .Debug;
+    const trace_init = if (want_tracing) std.debug.Trace(1, 3){} else {};
+    const Trace = @TypeOf(trace_init);
```

Comment:
It feels like this would be a common enough use case to justify adding a wrapper next to `std.debug.Trace` so that users don't have to reimplement it every time.

Reply:
The whole thing? I was actually thinking of changing it to be `const want_tracing = build_options.tracing;`, depending on a `-Dtracing` flag passed to `zig build`. Also the choice of 1 trace and 3 stack frames is a bit application-specific. But I see your point :thinking: :thinking: 

OK I think we can provide such a thing with decent defaults, and we can improve on it too - instead of `void` it could be an empty struct that provides the same methods, which are no-ops.

Thanks for the push!

Reply:
> The whole thing?

I meant something like `fn DebugOnlyTrace(comptime size: usize, comptime stack_frame_count: usize) type` which would have the same methods just like you mentioned.

> I was actually thinking of changing it to be `const want_tracing = build_options.tracing;`, depending on a `-Dtracing` flag passed to `zig build`.

Hmm, I guess we could also have `fn Trace(comptime size: usize, comptime stack_frame_count: usize) type` and `fn OptionalTrace(comptime size: usize, comptime stack_frame_count: usize, comptime enabled: bool) type` where `Trace` just returns an `OptionalTrace` with `enable` being `mode == .Debug`?